lib/HistoryModelService.js

```diff
@@ -5,90 +5,64 @@ var HistoryModelFields = require('./HistoryModelFields');
 var WaterlineLifecycleCallback = require('./WaterlineLifecycleCallback');
 
 var self = {
-  createModels: function createModels(sails) {
-    async.auto({
-      loadCollection: createAllModelHistory,
-      loadModel: ['loadCollection', loadModels]
-    }, function(err, result) {
-      if (err) {
-        sails.log.debug('Error creating history models ', err);
-      } else {
-        sails.log.info('Successful creating of history models');
-      }
-    });
+  createModels: function createModels(sails, callback){
+    try {
+      async.auto({
+        loadCollection: createAllModelHistory,
+        tearDown: ['loadCollection', function tearDatabaae(asyncCb) {
+          self.tearDownDatabase(self.getConnections(), asyncCb);
+        }],
+        loadModel: ['tearDown', loadModels]
+      }, function(err, result) {
+        if (err) {
+          sails.log.error('Error creating history models ', err);
+          callback(err);
+        } else {
+          sails.log.info('Successful creating of history models');
+          callback(null, true);
+        }
+      });
+    } catch(e) {
+      sails.log.error('Error creating history models');
+      callback(e);
+    }
 
-    function createAllModelHistory(asyncCb, asyncRes) {
+    function createAllModelHistory(asyncCb) {
```

Comment:
No need to extract into an internal function. May just specify this inside the async.auto.